servers/lib/server/handlers/api/gitlab/user.coffee

```diff
@@ -0,0 +1,150 @@
+hat = require 'hat'
+async = require 'async'
+
+apiErrors = require '../errors'
+{ handleUsername } = require '../helpers'
+
+GenericHandler = require './generichandler'
+
+
+module.exports = class User extends GenericHandler
+
+  # event user_create
+  @create = (data, callback = -> ) ->
+
+    # validating req params
+    { error, username, email, firstName,
+      lastName, suggestedUsername } = @validateDataFor 'create', data
+
+    return callback error  if error
+
+    bongo     = @getBongo()
+    { JUser } = @getModels()
+
+    client = null
+
+    queue  = [
+
+      (next) ->
+        JUser.one { username }, (err, user) ->
+          if err or user
+            return next err ? { message: 'user already exists' }
+          next()
+
+      (next) ->
+        handleUsername username, suggestedUsername, (err, _username) ->
+          return next err  if err
+          username = _username
+          next()
+
+      (next) ->
+        context = { group: 'gitlab' }
+        bongo.fetchClient 1, context, (client_) ->
```

Comment:
having 1 here is problematic. Could you pass clientId from `req` into this func? ( better to do smth like generateFakeClientFromReq does)